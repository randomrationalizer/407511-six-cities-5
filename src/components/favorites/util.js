export const getCitiesFromOffers = (offers) => {
  const cities = offers.slice().map((offer) => offer.city.name).sort();
  return [...new Set(cities)];
};
