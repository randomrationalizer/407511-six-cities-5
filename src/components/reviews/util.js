const REVIEW_MIN_LENGTH = 50;
const REVIEW_MAX_LENGTH = 300;
const RATING_MIN = 1;
const RATING_MAX = 5;

export const checkCommentValidity = (comment) => {
  return comment.length >= REVIEW_MIN_LENGTH && comment.length <= REVIEW_MAX_LENGTH ? true : false;
};

export const checkRatingValidity = (rating) => {
  return rating >= RATING_MIN && rating <= RATING_MAX ? true : false;
};
