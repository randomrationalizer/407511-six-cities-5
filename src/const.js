export const cities = [
  {
    name: `Paris`,
    coords: {
      latitude: 48.864716,
      longitude: 2.349014
    }
  },
  {
    name: `Cologne`,
    coords: {
      latitude: 50.935173,
      longitude: 6.953101
    }
  },
  {
    name: `Brussels`,
    coords: {
      latitude: 50.8505,
      longitude: 4.3488
    }
  },
  {
    name: `Amsterdam`,
    coords: {
      latitude: 52.38333,
      longitude: 4.9
    }
  },
  {
    name: `Hamburg`,
    coords: {
      latitude: 53.551086,
      longitude: 9.993682
    }
  },
  {
    name: `Dusseldorf`,
    coords: {
      latitude: 51.233334,
      longitude: 6.783333
    }
  }
];

export const OfferType = {
  MAIN: `cities`,
  NEARBY: `near-places`,
  FAVORITES: `favorites`
};

export const MapType = {
  MAIN: `cities`,
  PROPERTY: `property`
};

export const SortType = {
  DEFAULT: `DEFAULT`,
  PRICE_UP: `PRICE_UP`,
  PRICE_DOWN: `PRICE_DOWN`,
  RATING_DOWN: `RATING_DOWN`
};
