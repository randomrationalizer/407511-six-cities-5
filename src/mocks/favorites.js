export const favorites = [`1`, `3`, `4`];
