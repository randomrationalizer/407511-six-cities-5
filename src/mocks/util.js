import configureStore from "redux-mock-store";

export const noop = () => {};

export const getMockStore = (store) => {
  const storeConfig = configureStore();
  return storeConfig(store);
};

export const nope = () => {
  return Promise.resolve({});
};
