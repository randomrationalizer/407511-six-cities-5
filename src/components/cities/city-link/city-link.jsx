import React from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";
import {changeCity} from "../../../store/action";
import {CityLinkType} from "../../../const";

const getClassName = {
  [CityLinkType.MAIN_PAGE]: (isActive) => `locations__item-link tabs__item ${isActive ? `tabs__item--active` : ``}`,
  [CityLinkType.INNER_PAGE]: () => `locations__item-link`
};


const CityLink = ({city, isActive, onCityChange, linkType}) => {
  const isMainPageLink = linkType === CityLinkType.MAIN_PAGE;
  const linkClassName = isMainPageLink ? getClassName[linkType](isActive) : getClassName[linkType]();


  const handleClick = () => {
    if (isMainPageLink && isActive) {
      return;
    }

    onCityChange(city);
  };

  return (
    <Link to={`/#${city.toLowerCase()}`}
      onClick={handleClick}
      className={linkClassName}
    >
      <span>{city}</span>
    </Link>
  );
};

CityLink.propTypes = {
  city: PropTypes.string.isRequired,
  isActive: PropTypes.bool,
  onCityChange: PropTypes.func.isRequired,
  linkType: PropTypes.string.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onCityChange(city) {
    dispatch(changeCity(city));
  }
});


export {CityLink};
export default connect(null, mapDispatchToProps)(CityLink);
