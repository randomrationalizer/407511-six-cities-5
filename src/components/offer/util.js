import dayjs from "dayjs";
import {OfferType} from "../../const";


export const capitalize = (str) => {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

// Возвращает дату в виде строки в формате: 2019-04-24
export const formatDate = (date) => {
  return dayjs(date).format(`YYYY-MM-DD`);
};

// Возвращает дату в виде строки в формате: January 2020
export const humanizeDate = (date) => {
  return dayjs(date).format(`MMMM YYYY`);
};

// Сравнение для сортировки отзывов в хронологическом порядке
export const sortByDate = (a, b) => {
  return dayjs(a.date) - dayjs(b.date);
};

export const getDescriptionSentences = (text) => {
  let description = text.split(`. `);
  description[description.length - 1] = description[description.length - 1].replace(`.`, ``);

  return description;
};

export const isFavoritesCard = (type) => {
  return type === OfferType.FAVORITES;
};

export const isMainPageCard = (type) => {
  return type === OfferType.MAIN;
};
