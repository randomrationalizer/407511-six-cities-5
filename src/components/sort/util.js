import {SortType} from "../../const";

export const sortTypeToTitle = {
  [SortType.DEFAULT]: `Popular`,
  [SortType.PRICE_UP]: `Price: low to high`,
  [SortType.PRICE_DOWN]: `Price: high to low`,
  [SortType.RATING_DOWN]: `Top rated first`
};
