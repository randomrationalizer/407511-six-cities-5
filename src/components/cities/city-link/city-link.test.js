import React from "react";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {CityLink} from "./city-link";
import {DEFAULT_CITY, CityLinkType} from "../../../const";
import {noop} from "../../../mocks/util";


describe(`Should City link component renders correctly`, () => {
  describe(`for main page`, () => {
    it(`with active link`, () => {
      const tree = renderer
      .create(
          <BrowserRouter>
            <CityLink
              city={DEFAULT_CITY}
              isActive={true}
              onCityChange={noop}
              linkType={CityLinkType.MAIN_PAGE}
            />
          </BrowserRouter>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`with inactive link`, () => {
      const tree = renderer
      .create(
          <BrowserRouter>
            <CityLink
              city={DEFAULT_CITY}
              isActive={false}
              onCityChange={noop}
              linkType={CityLinkType.MAIN_PAGE}
            />
          </BrowserRouter>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  it(`for inner pages`, () => {
    const tree = renderer
    .create(
        <BrowserRouter>
          <CityLink
            city={DEFAULT_CITY}
            onCityChange={noop}
            linkType={CityLinkType.INNER_PAGE}
          />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
