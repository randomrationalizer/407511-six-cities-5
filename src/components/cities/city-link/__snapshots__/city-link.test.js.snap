// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should City link component renders correctly for inner pages 1`] = `
<a
  className="locations__item-link"
  href="/#paris"
  onClick={[Function]}
>
  <span>
    Paris
  </span>
</a>
`;

exports[`Should City link component renders correctly for main page with active link 1`] = `
<a
  className="locations__item-link tabs__item tabs__item--active"
  href="/#paris"
  onClick={[Function]}
>
  <span>
    Paris
  </span>
</a>
`;

exports[`Should City link component renders correctly for main page with inactive link 1`] = `
<a
  className="locations__item-link tabs__item "
  href="/#paris"
  onClick={[Function]}
>
  <span>
    Paris
  </span>
</a>
`;
